sensor/common/detector/enricher.go

```diff
@@ -356,6 +355,16 @@ func (e *enricher) getImages(deployment *storage.Deployment) []*storage.Image {
 	return images
 }
 
+func (e *enricher) getPullSecrets(deployment *storage.Deployment) []string {
+	var pullSecrets []string
```

Comment:
A set was used however that was destructive to the order defined in the spec (which k8s respects). This will return the pull secret names as defined in k8s, so there will be duplicates secret names if k8s allows duplicates to be specified in the podspec and/or service account.

The kubelet [does not appear to filter](https://github.com/kubernetes/kubernetes/blob/62f96d2748285eb556705af7510d0065ff84a7b1/pkg/kubelet/kubelet_pods.go#L999-L1013) repeated entries